### ISSUE TITLE
使用material_.showDatePicker后内存飙升，卡死

### ISSUE THREAD

**Issue by tomcatter** [Unity-Technologies/com.unity.uiwidgets#280]
我在使用`material_.showDatePicker` 遇到两个问题

**一**：是[`calendar_date_picker.cs`](https://github.com/Unity-Technologies/com.unity.uiwidgets/blob/master/com.unity.uiwidgets/Runtime/material/pickers/calendar_date_picker.cs) 中有变量赋值错误：
```
  initialDate = utils.dateOnly(initialDate.Value);
  firstDate = utils.dateOnly(firstDate.Value);
  lastDate = utils.dateOnly(lastDate.Value);
```
导致在对应的后面处理取到的值有问题，我将这三处改成了
```
  this.initialDate = utils.dateOnly(initialDate.Value);
  this.firstDate = utils.dateOnly(firstDate.Value);
  this.lastDate = utils.dateOnly(lastDate.Value);
```

**二**：将上面的问题修改以后运行，内存飙升，很快就到了`100%`，整个电脑卡死

**Comment by zhuxingwei** [comment]
Hi Thanks for the kind report !

Let me look into this issue first. Thanks a lot for your **great helps** to UIWidgets !

**Comment by zhuxingwei** [comment]
fixed by #279

**Comment by zhuxingwei** [comment]
fixed by #289

**Comment by tomcatter** [comment]
great job, this issue has done.
